torch/nn/modules/transformer.py

```diff
@@ -259,21 +272,36 @@ class TransformerEncoderLayer(Module):
         >>> out = encoder_layer(src)
     """
 
-    def __init__(self, d_model, nhead, dim_feedforward=2048, dropout=0.1, activation="relu"):
+    def __init__(self, d_model, nhead, dim_feedforward=2048, dropout=0.1, activation="relu",
+                 *, reset_parameters: bool = True, **kwargs) -> None:
+        factory_kwargs = torch.factory_kwargs(kwargs)
         super(TransformerEncoderLayer, self).__init__()
-        self.self_attn = MultiheadAttention(d_model, nhead, dropout=dropout)
+        self.self_attn = MultiheadAttention(d_model, nhead, dropout=dropout,
+                                            reset_parameters=reset_parameters, **factory_kwargs)
         # Implementation of Feedforward model
-        self.linear1 = Linear(d_model, dim_feedforward)
+        self.linear1 = Linear(d_model, dim_feedforward,
+                              reset_parameters=reset_parameters, **factory_kwargs)
         self.dropout = Dropout(dropout)
-        self.linear2 = Linear(dim_feedforward, d_model)
+        self.linear2 = Linear(dim_feedforward, d_model,
+                              reset_parameters=reset_parameters, **factory_kwargs)
 
-        self.norm1 = LayerNorm(d_model)
-        self.norm2 = LayerNorm(d_model)
+        self.norm1 = LayerNorm(d_model, reset_parameters=reset_parameters, **factory_kwargs)
+        self.norm2 = LayerNorm(d_model, reset_parameters=reset_parameters, **factory_kwargs)
         self.dropout1 = Dropout(dropout)
         self.dropout2 = Dropout(dropout)
 
         self.activation = _get_activation_fn(activation)
 
+        if reset_parameters:
+            self.reset_parameters()
```

Comment:
Probably better not to have this one either

Reply:
Fixed